lldb/test/API/lang/swift/async/unwind/unwind_in_all_instructions/TestSwiftAsyncUnwindAllInstructions.py

```diff
@@ -0,0 +1,134 @@
+    # see rdar://137048317. For now, we only know how to unwind during and
+    # shortly after the prologue. This function returns "should skip" if we're
+    # at a PC that is too far from the prologue (~16 bytes). This is a
+    # rough approximation that seems to work for both x86 and arm.
+    def should_skip_Q_funclet(self, thread):
+        current_frame = thread.frames[0]
+        function = current_frame.GetFunction()
+        if "await resume" not in function.GetName():
+            return False
+
+        max_prologue_offset = 16
+        prologue_end = function.GetStartAddress()
+        prologue_end.OffsetAddress(function.GetPrologueByteSize() + max_prologue_offset)
+        current_pc = current_frame.GetPCAddress()
+        return current_pc.GetFileAddress() >= prologue_end.GetFileAddress()
+
+    def check_unwind_ok(self, thread, bpid):
+        if self.should_skip_Q_funclet(thread):
+            return
+        # Check that we see the virtual backtrace:
+        expected_funcnames = [
+            "ASYNC___1___",
+            "ASYNC___2___",
+            "ASYNC___3___",
+            "ASYNC___4___",
+            "ASYNC___5___",
+        ]
+        frames = thread.frames
+        self.assertGreater(
+            len(frames), len(expected_funcnames), f"Invalid backtrace for {frames}"
+        )
+        actual_funcnames = [
+            frame.GetFunctionName() for frame in frames[: len(expected_funcnames)]
+        ]
+        for expected_name, actual_name in zip(expected_funcnames, actual_funcnames):
+            self.assertIn(expected_name, actual_name, f"Unexpected backtrace: {frames}")
+
+    @swiftTest
+    @skipIf(oslist=["windows", "linux"])
+    def test(self):
+        """Test that the debugger can unwind at all instructions of all funclets"""
+        self.build()
+
+        source_file = lldb.SBFileSpec("main.swift")
+        target, process, _, _ = lldbutil.run_to_source_breakpoint(
+            self, "BREAK HERE", source_file
+        )
+
+        breakpoints = self.set_breakpoints_all_funclets(target)
+        num_breakpoints = len(breakpoints)
+
+        # Reach most breakpoints and ensure we can unwind in that position.
+        while True:
+            process.Continue()
+            thread = lldbutil.get_stopped_thread(process, lldb.eStopReasonBreakpoint)
+            if thread is None:
+                break
+            bpid = thread.GetStopReasonDataAtIndex(0)
+            breakpoints.remove(bpid)
+            target.FindBreakpointByID(bpid).SetEnabled(False)
+
+            self.check_unwind_ok(thread, bpid)
+
+        # We will never hit all breakpoints we set, because of things like
+        # overflow handling or other unreachable traps. However, it's good to
+        # have some sanity check that we have hit at least a decent chunk of
+        # them.
+        breakpoints_not_hit = len(breakpoints)
+        self.assertLess(breakpoints_not_hit / num_breakpoints, 0.10)
```

Comment:
to follow up here, what's the ratio now?

Reply:
It is 6% today. Do you think this is too close to the 10%?

I was honestly debating whether to even include this check, my original motivation was that if somehow we never hit any breakpoints, then this test would still pass. This check would prevent something like that from going undetected, but it's not that elegant.

Reply:
I was mostly curious. 10% seems reasonable. It may not elegant, but could prove practical. It's probably better to start with this than without.